### ISSUE TITLE
[BUG] Cannot instantiate interface ConfigurationManagerInterface

### ISSUE THREAD

**Issue by chroenmeister** [TYPO3-Solr/ext-solr#3622]
**Describe the bug**
Using the latest 11.2.2 release getting this exception while opening plugin instance:

Cannot instantiate interface TYPO3\CMS\Extbase\Configuration\ConfigurationManagerInterface

**To Reproduce**
1. Open Plugin Search: Form, Result, Additional Components

**Screenshots**

![grafik](https://user-images.githubusercontent.com/5887116/235876303-c35b851e-7e31-4290-bea8-0806eb108eda.png)

**Used versions (please complete the following information):**
 - TYPO3 Version: 10.4.37
 - EXT:solr Version: 11.2.2
 - PHP Version: 7.4.11

**Comment by brandung-sjorek** [comment]
The same applies to:

- TYPO3 Version: 11.5.26
- EXT:solr Version: **11.5.2**
- PHP Version: 7.4.11

**Comment by dkd-kaehm** [comment]
We can not reproduce the issue.

@lfehr, @chroenmeister, @brandung-sjorek 
Please provide the installed extension list, or better those, which implementing the classes   
`TYPO3\CMS\Extbase\Configuration\ConfigurationManagerInterface`.

Most probably, there is some extension which implements this interface and therefore the DI can not resolve the class.

**Comment by brandung-sjorek** [comment]
@dkd-kaehm We'll (@chroenmeister and me) look into it, and give you Feedback within the next 2 weeks. Please keep the issue open, in the case we find the root-cause in the TYPO3 Solr-Extension itself.

Thanks for your reply …

**Comment by rr-it** [comment]
https://github.com/TYPO3-Solr/ext-solr/blob/3fc0a33f627530eb9b7c4ad0b6a7f7e894977467/Classes/System/UserFunctions/FlexFormUserFunctions.php#L206-L216

We are trying to make an instance of an `Interface::class` here. Not sure if this must be an instance of an actual implementation of this interface?

```php
 $configurationManager = GeneralUtility::makeInstance(\ApacheSolrForTypo3\Solr\System\Configuration\ConfigurationManager::class);
$configurationManager = GeneralUtility::makeInstance(\TYPO3\CMS\Extbase\Configuration\ConfigurationManager::class);
$configurationManager = GeneralUtility::makeInstance(\TYPO3\CMS\Core\Configuration\ConfigurationManager::class);
```


And `$pid` is not used anymore to get the paged-based configuration. Is this expected behaviour?

---

Usually the `ConfigurationManager` is injected like: 

```php
    /**
     * @var \TYPO3\CMS\Extbase\Configuration\ConfigurationManagerInterface
     */
    protected $configurationManager;

    /**
     * Injects the Configuration Manager and loads the settings
     *
     * @param \TYPO3\CMS\Extbase\Configuration\ConfigurationManagerInterface $configurationManager An instance of the Configuration Manager
     */
    public function injectConfigurationManager(
        \TYPO3\CMS\Extbase\Configuration\ConfigurationManagerInterface $configurationManager
    ) {
        $this->configurationManager = $configurationManager;
    }
```

**Comment by rr-it** [comment]
> **To Reproduce**
> 
>     1. Open Plugin Search: Form, Result, Additional Components
> 

1. In **TYPO3 Backend** go to the module "WEB" --> "Page"
2. Edit a content element of type "Plugin Search"

**Comment by brandung-sjorek** [comment]
@rr-it @dkd-kaehm The ConfigurationManager must be injected via the `TYPO3\CMS\Extbase\Configuration\ConfigurationManagerInterface` , otherwise the DI is not able to resolve the correct implementation (`TYPO3\CMS\Extbase\Configuration\BackendConfigurationManager` or `TYPO3\CMS\Extbase\Configuration\BackendConfigurationManager`)

**Comment by dkd-kaehm** [comment]
@rr-it 
Tried to reproduce this issue within https://github.com/TYPO3-Solr/solr-ddev-site 11.5.x and 11.2.x.

---
@brandung-sjorek Please kick the pull request for this issue.

**Comment by brandung-sjorek** [comment]
@rr-it Forgot to say: Thanks for your investigation …

@dkd-kaehm Which pull-request? Maybe I get something wrong - do you want us to create a PR? No problem, but the colleague will need some time (the 14 days mentioned above)

**Comment by rr-it** [comment]
Some points on the reproducibility of this bug:
There might be a "race condition" or similar introduced by the use of  `GeneralUtility::makeInstance()`.

In case an object matching `ConfigurationManagerInterface::class` has been successfully instantiated before by some other extension, this object is returned perfectly fine from `GeneralUtility::makeInstance()` to use by Solr `getConfigurationFromPageId()`.

If Solr `getConfigurationFromPageId()` is first to instantiate `ConfigurationManagerInterface::class` via direct call to `GeneralUtility::makeInstance()`  it fails.

https://github.com/TYPO3/typo3/blob/808ef61a53bc1e7ef65b715c13727b0f2d086c24/typo3/sysext/core/Classes/Utility/GeneralUtility.php#L3465C1-L3489

**Comment by dkd-kaehm** [comment]
Seems to be fixed for TYPO3 10 and 11 in #3626 and #3640.

@chroenmeister Please check latest state of release-11.2.x branch and close this issue if it is fixed.

**Comment by chroenmeister** [comment]
@dkd-kaehm Yes the issue is solved in the latest release-11.2.x branch! Thank you!
